app/uk/gov/hmrc/payetaxcalculatorfrontend/controllers/QuickCalcController.scala

```diff
@@ -82,25 +82,52 @@ class QuickCalcController @Inject()(override val messagesApi: MessagesApi, cache
     val url = request.uri
     Salary.salaryBaseForm.bindFromRequest().fold(
       formWithErrors => Future(BadRequest(salary(formWithErrors))),
-
       salaryAmount => {
-        val updatedAggregate = cache.fetchAndGetEntry()
-          .map(_.getOrElse(QuickCalcAggregateInput.newInstance))
-          .map(_.copy(savedSalary = Some(salaryAmount), savedPeriod = None))
+        val updatedAggregate = updateSalaryAmount(salaryAmount, url)
 
         updatedAggregate.flatMap(agg => cache.save(agg).map( _ => {
           salaryAmount.period match {
-            case "a day" => Redirect(routes.QuickCalcController.showDaysAWeek(Salary.salaryInPence(salaryAmount.amount), url))
-            case "an hour" => Redirect(routes.QuickCalcController.showHoursAWeek(Salary.salaryInPence(salaryAmount.amount), url))
-            case _ => nextPageOrSummaryIfAllQuestionsAnswered(agg){
-              Redirect(routes.QuickCalcController.showStatePensionForm())
-            }
+            case "a day" =>
+              if(agg.savedPeriod.map(_.period).contains("a day")) tryGetShowStatePension(agg)
+              else Redirect(routes.QuickCalcController.showDaysAWeek(Salary.salaryInPence(salaryAmount.amount), url))
```

Comment:
import routes.QuickCalcController._ might be useful
Salary.salaryInPence(salaryAmount.amount) might be reused as val for both cases